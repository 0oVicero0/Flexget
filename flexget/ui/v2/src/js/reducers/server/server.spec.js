import reducer from 'reducers/server';
import {
  SERVER_RELOAD,
  SERVER_SHUTDOWN,
  SERVER_RELOAD_DISMISS,
  SERVER_SHUTDOWN_PROMPT,
  SERVER_SHUTDOWN_PROMPT_DISMISS,
  SERVER_SHUTDOWN_DISMISS,
} from 'actions/server';

describe('reducers/server', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('reload should be true on SERVER_RELOAD', () => {
    expect(reducer(undefined, { type: SERVER_RELOAD }).reload).toBe(true);
  });

  it('reload should be true on SERVER_RELOAD_DISMISS', () => {
    expect(reducer(undefined, { type: SERVER_RELOAD_DISMISS }).reload).toBe(false);
  });

  it('reload should be true on SERVER_SHUTDOWN', () => {
    expect(reducer(undefined, { type: SERVER_SHUTDOWN }).shutdown).toBe(true);
  });

  it('reload should be true on SERVER_SHUTDOWN_DISMISS', () => {
    expect(reducer(undefined, { type: SERVER_SHUTDOWN_DISMISS }).shutdown).toBe(false);
  });

  it('reload should be true on SERVER_SHUTDOWN_PROMPT', () => {
    expect(reducer(undefined, { type: SERVER_SHUTDOWN_PROMPT }).shutdownPrompt).toBe(true);
  });

  it('reload should be true on SERVER_SHUTDOWN_DISMISS', () => {
    expect(reducer(undefined, { type: SERVER_SHUTDOWN_PROMPT_DISMISS }).shutdownPrompt).toBe(false);
  });
});
