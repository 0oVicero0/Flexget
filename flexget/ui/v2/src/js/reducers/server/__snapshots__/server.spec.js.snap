// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/server should return the initial state 1`] = `
Object {
  "reload": false,
  "shutdown": false,
  "shutdownPrompt": false,
}
`;
